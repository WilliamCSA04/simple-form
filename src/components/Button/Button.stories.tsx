import * as React from 'react';
import Button from '.';
import { GlobalStyles } from '../../styles';

export default {
  title: 'Button',
  decorators: [
    (storyFn) => (
      <>
        <GlobalStyles />
        {storyFn()}
      </>
    ),
  ],
};

export const Default = () => <Button onClick={(e) => e.preventDefault}>Press me</Button>;
