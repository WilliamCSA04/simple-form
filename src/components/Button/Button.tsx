import * as React from 'react';
import styled from 'styled-components';

export interface Props {
  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
  children?: React.ReactNode;
  id?: string;
}

const StyledButton = styled.button`
  padding: 0.5em;
  border-radius: 3px;
  border: 0;
  background-color: #b0e298;
  color: white;
  box-sizing: border-box;
`;

export default function Button({ id, onClick, children }: Props) {
  return (
    <StyledButton id={id} onClick={onClick}>
      {children}
    </StyledButton>
  );
}
