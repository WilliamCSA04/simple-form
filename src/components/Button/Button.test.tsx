import React from 'react';
import { render } from '@testing-library/react';
import Button from '.';

describe('<Button /> snapshot', () => {
  it('renders the component with default props', () => {
    const container = render(<Button id="Test" onClick={() => {}} />);
    expect(container.asFragment()).toMatchSnapshot();
  });
});
