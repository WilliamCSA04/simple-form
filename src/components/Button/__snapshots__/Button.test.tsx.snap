// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot renders the component with default props 1`] = `
<DocumentFragment>
  <button
    id="Test"
  />
</DocumentFragment>
`;
